### PR TITLE
refactor(dataframe): remove index in "Dataframe" and "column.Series"

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-28v2d3k 实现 DataFrame.extend 方法](https://app.clickup.com/t/28v2d3k)

File: graviti/dataframe/column/indexing.py [hunk 1]
```diff
@@ -21,15 +21,15 @@ def __init__(self, obj: "ColumnSeries") -> None:
     # def __getitem__(self, key: Union[Iterable[bool], slice]) -> "ColumnSeries":
     #    ...
 
-    @overload
-    def __getitem__(self, key: int) -> Any:
-        ...
+    # @overload
+    # def __getitem__(self, key: int) -> Any:
+    #     ...
```
Comment (linjiX) [inline graviti/dataframe/column/indexing.py:26]: The commit message is not clear

File: graviti/dataframe/column/series.py [hunk 2]
```diff
@@ -43,54 +43,38 @@ class Series:
...
+    # def __getitem__(self, key: int) -> Any:
+    #     ...
+    #
+    # @overload
+    # def __getitem__(self, key: Iterable[int]) -> "Series":
+    #     ...
 
-        return self._data[integer_location]
+    def __getitem__(self, key: int) -> Any:
+        return self._data[key].as_py()
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:77]: do we need to return after `as_py`?

File: graviti/dataframe/column/series.py [hunk 3]
```diff
@@ -43,54 +43,38 @@ class Series:
...
         self._data = data
         self.name = name
         self.schema = schema
-        if index is None:
-            self._indices_data = index
-            self._indices = index
-        else:
-            self._indices_data = {raw_index: location for location, raw_index in enumerate(index)}
-            self._indices = list(index)
+        self._indices = range(len(self))
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:62]: why we need `self._indices`?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**